dbee/clients/common/client.go

```diff
@@ -8,7 +8,20 @@ import (
 	"github.com/kndndrj/nvim-dbee/dbee/models"
 )
 
-// default sql client used by other specific implementations
+// DatabaseClient is an interface that represents a database client.
+type DatabaseClient interface {
+	Conn() (DatabaseConnection, error)
+	Close()
+	Swap(db *sql.DB)
+}
+
+// DatabaseConnection is an interface that represents a database connection.
+type DatabaseConnection interface {
+	Close() error
+	Exec(query string) (models.IterResult, error)
+	Query(query string) (models.IterResult, error)
+}
+
 type Client struct {
```

Comment:
The point of these `common` files is to provide a result builder and a client that works for "most" drivers. If the driver's functionality is too specific, i think it's better that it uses it's own implementation.